onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -1334,7 +1334,49 @@ static constexpr OrtApiBase ort_api_base = {
     &OrtApis::GetVersionString,
 };
 
-static constexpr OrtApi ort_api_1 = {
+/* Rules on how to add a new Ort API version
+
+In general, NEVER remove or rearrange the members in this structure unless a new version is being created. The
+goal is for newer shared libraries of the Onnx Runtime to work with binaries targeting the previous versions.
+In order to do that we need to ensure older binaries get the older interfaces they are expecting.
+
+If the next version of the OrtApi only adds members, new members can be added at the end of the OrtApi structure
+without breaking anything. In this case, rename the ort_api_# structure in a way that shows the range of versions
+it supports, for example 'ort_api_1_to_2', and then GetApi can return the same structure for a range of versions.
+
+If methods need to be removed or rearranged, then make a copy of the OrtApi structure and name it 'OrtApi#to#'.
+The latest Api should always be named just OrtApi. Then make a copy of the latest ort_api_* structure below and
+name it ort_api_# to match the latest version number supported, you'll need to be sure the structure types match
+the API they're for (the compiler should complain if this isn't correct).
+
+If there is no desire to have the headers still expose the older APIs (clutter, documentation, etc) then the
+definition should be moved to a file included by this file so that it's still defined here for binary compatibility
+but isn't visible in public headers.
+
+So for example, if we wanted to just add some new members to the ort_api_1_to_2, we'd take the following steps:
+
+	In include\onnxruntime\core\session\onnxruntime_c_api.h we'd just add the members to the end of the structure
+
+	In this file, we'd correspondingly add the member values to the end of the ort_api_1_to_2 structure, and also rename
+	it to ort_api_1_to_3.
+
+	Then in GetApi we'd make it return ort_api_1_to_3 for versions 1 through 3.
+
+Second example, if we wanted to add and remove some members, we'd do this:
+
+	In include\onnxruntime\core\session\onnxruntime_c_api.h we'd make a copy of the OrtApi structure and name the
+	old one OrtApi1. In the new OrtApi we'd add or remove any members that we desire.
+
```

Comment:
OrtApi1 or OrtApi1to2 given it was handling both version right?

Reply:
Yep! I wrote it originally without there being a version 2, so I missed that part.